### ISSUE TITLE
Rule proposal: Forbid `!foo === bar`

### ISSUE THREAD

**Issue by fisker** [sindresorhus/eslint-plugin-unicorn#1421]
I'm not sure what name it should be, this is a common mistake, espeically when doing quick debugging. Personally I've done this several times. 

We have this syntax forbidden in Prettier codebase, with [`no-restricted-syntax` rule](https://github.com/prettier/prettier/blob/c2599d47145f7b4bd310f66edd69d9ca14b2c60d/.eslintrc.js#L38-L39), because I made [a mistake](https://github.com/prettier/prettier/commit/c7e354e38cfd051d3f64b56aeda35e9a0a669de6#r47946398) once there.

## Fail

```js
if (!foo === bar);
```

```js
if (!foo !== bar);
```

```js
// Don't have to be in `if`
const isEqual = !foo === bar;
```

## Pass

```js
if (!(foo === bar));
```

```js
// I guess we should allow this, it's type-casting
if (!!foo === bar);
```

**Comment by sindresorhus** [comment]
`no-negation-in-equality-check` ?

**Comment by sindresorhus** [comment]
This looks similar to https://eslint.org/docs/rules/no-unsafe-negation Maybe that rule could support this too?

**Comment by fisker** [comment]
That rule not checking this case, but I can make a proposal there.

**Comment by fisker** [comment]
Forgot to mention here, issue on ESLint codebase https://github.com/eslint/eslint/issues/14802, and seems they are going accept.

**Comment by fregante** [comment]
`no-mixed-operators` can already do this, if configured correctly.

- https://github.com/xojs/eslint-config-xo/issues/73

**Comment by fisker** [comment]
Maybe we should also forbid `!foo in bar` and `!foo instanceof bar`.

Rule name `no-negation-in-equality-check` may won't fit.

**Comment by fisker** [comment]
`no-negated-binary-expression-left`?

**Comment by fisker** [comment]
Accepted.
